elements/pl-order-blocks/dag_checker.py

```diff
@@ -45,27 +46,92 @@ def check_grouping(order: list[str], group_belonging: Mapping[str, Optional[int]
     return len(order)
 
 
-def grade_dag(order: list[str], depends_graph: Mapping[str, list[str]], group_belonging: Mapping[str, Optional[int]]) -> tuple[int, int]:
+def dag_to_nx(depends_graph: Mapping[str, list[str]]) -> nx.DiGraph:
+    """Convert input graph format from HTML into NetworkX object to utilize their algorithms."""
+    graph = nx.DiGraph()
+    for node in depends_graph:
+        for node2 in depends_graph[node]:
+            graph.add_edge(node2, node)
```

Comment:
To clarify understanding: `depends_graph` is a dict of reverse edges? So if the graph is `A -> B`, `depends_graph` would be `{'B': ['A']}`?

Reply:
Ya... when writing the questions as an instructor it makes more sense to think about the edges one way, but when I thought up the grading algorithms it was easier to think about them the other way. I guess it would only take changing a couple lines of code to make the grading algorithms work with the edges the other direction. Do you think it would significantly increase code readability if I did that?

Reply:
After a lot of thinking, and discussing with @mwest1066, I'm going to go ahead and leave this the way that it is. The reason why is because if I flip the edges the other way for the algorithm, then: 
- the diagrams of graphs I show in my paper would have to have the edges pointing upward instead of downward through the proof, which is pretty unconventional
- correct solutions to a proof would then become _reverse_ topological sorts instead of topological sorts.

I think both of these things together would cause more confusion than the current way of doing things does.

Instead, I will add a comment here clarifying that the depends graph lists the _incoming_ edges of a given node.